### ISSUE TITLE
Improve C analyse_text

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#855]
(Original issue 1148 created by None on 2015-09-11T12:25:02.556116+00:00)

This code snippet

```
#!c

#ifndef __foo__
#define __foo__

typedef struct {
  ...
} SomeStruct;

#endif
```

is recognized as Objective C by `get_lexer_for_filename("foo.h", code)` but should be recognized as C instead. The C `analyse_text` function could be extended to look for `#ifndef` to cover this case.

**Comment by Anteru** [comment]
(Original issue was assigned to tshatch)

(Original issue 1148 last updated on 2015-10-17T01:51:28.405704+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2015-10-17T01:51:16.608725+00:00)

Fixes #1148: Trigger using CLexer off string "#ifndef"

→ <<cset 3c3a3cbc44c9>>
